src/graphs/BasicGraphs.jl

```diff
@@ -101,15 +102,28 @@ edges(g::HasGraph) = parts(g, :E)
 edges(g::HasGraph, src::Int, tgt::Int) =
   (e for e in incident(g, src, :src) if subpart(g, e, :tgt) == tgt)
 
+""" Edges coming out of a vertex
+"""
+@inline outedges(g::HasGraph, v) = incident(g, v, :src)
+
+""" Edges coming into a vertex
+"""
+inedges(g::HasGraph, v) = incident(g, v, :tgt)
+
 """ Whether the graph has the given vertex.
 """
 has_vertex(g::HasVertices, v) = has_part(g, :V, v)
 
 """ Whether the graph has the given edge, or an edge between two vertices.
 """
-has_edge(g::HasGraph, e) = has_part(g, :E, e)
-has_edge(g::HasGraph, src::Int, tgt::Int) =
-  has_vertex(g, src) && tgt ∈ outneighbors(g, src)
+@inline has_edge(g::HasGraph, e) = has_part(g, :E, e)
+@inline function has_edge(g::HasGraph, s::Int, t::Int)
```

Comment:
Are these `@inline`s necessary? Based on my experience, I would guess no, but I'm not completely.

Also, while it may be beyond the scope of this PR, I think we should modify `outneighbors` and co to return iterators, which should avoid the need for longer function defintiion below.